typescript/cli/ci-test.sh

```diff
@@ -1,18 +1,47 @@
 #!/usr/bin/env bash
 
-# NOTE: This script is intended to be run from the root of the repo
-
-# the first arg to this script is a flag to enable the hook config as part of core deployment
-# motivation is to test both the bare bone deployment (included in the docs) and the deployment with the routing over igp hook (which is closer to production deployment)
-HOOK_FLAG=$1
-if [ -z "$HOOK_FLAG" ]; then
-  echo "Usage: fork.sh <hook>"
-  exit 1
+# set script location as repo root
+cd "$(dirname "$0")/../.."
+
+TEST_TYPE_PRESET_HOOK="preset_hook_enabled"
+TEST_TYPE_CONFIGURED_HOOK="configure_hook_enabled"
+TEST_TYPE_PI_CORE="pi_with_core_chain"
+
+# set the first arg to 'configured_hook' to set the hook config as part of core deployment
+# motivation is to test both the bare bone deployment (included in the docs) and the deployment
+# with the routing over igp hook (which is closer to production deployment)
+TEST_TYPE=$1
+if [ -z "$TEST_TYPE" ]; then
+    echo "Usage: ci-test.sh <test-type>"
+    exit 1
 fi
 
+HOOK_FLAG=false
+if [ "$TEST_TYPE" == $TEST_TYPE_CONFIGURED_HOOK ]; then
+    HOOK_FLAG=true
+fi
+
+CHAIN1=anvil1
+CHAIN2=anvil2
+EXAMPLES_PATH=./examples
+
+# use different chain names and config for pi<>core test
+if [ "$TEST_TYPE" == $TEST_TYPE_PI_CORE ]; then
+    CHAIN1=anvil
+    CHAIN2=ethereum
+    EXAMPLES_PATH=./examples/fork
+fi
+
+CHAIN1_CAPS=$(echo "${CHAIN1}" | tr '[:lower:]' '[:upper:]')
+CHAIN2_CAPS=$(echo "${CHAIN2}" | tr '[:lower:]' '[:upper:]')
```

Comment:
This script is getting more and more fancy. Wondering if it should be converted to JS soon